lib/App/DuckPAN/Web.pm

```diff
@@ -353,6 +357,7 @@ sub request {
 					my $structured = $result->structured_answer;
 					push @ids, $structured->{dynamic_id} if exists $structured->{dynamic_id};
 					if(exists $structured->{templates}){ # user-specified templates
+						$structured->{id} = $id if $id;
```

Comment:
@zachthompson I did that because `$id` wouldn't be defined above when Cheat Sheets are triggered, but it's OK because `$structured->{id}` is already defined for Cheat Sheets.